code/controllers/subsystem/aura.dm

```diff
@@ -0,0 +1,98 @@
+/// Tracks aura emitters and tells them to pulse. Also used for creating new ones.
+SUBSYSTEM_DEF(aura)
+	name = "Auras"
+	wait = 2 SECONDS
+	flags = SS_NO_INIT
+	///Stores currently active aura_bearer datums
+	var/list/active_auras = list() //We can't use a normal subsystem processing list because as soon as an aura_bearer leaves the list it needs to die
+
+/datum/controller/subsystem/aura/fire(resumed)
+	for(var/datum/aura_bearer/bearer AS in active_auras)
+		bearer.process()
+
+///Use this to start a new emitter with the specified stats. Returns the emitter in question, use stop_emitting() to end early.
+/datum/controller/subsystem/aura/proc/add_emitter(atom/center, type, range, strength, duration)
+	if(!istype(center))
+		return
+	if(!type || !range || !strength)
+		return
+	return new /datum/aura_bearer(center, type, range, strength, duration)
+
+///The thing that actually pushes out auras to nearby mobs.
+/datum/aura_bearer
+	///What we emit from
+	var/atom/center
+	///String indicating aura variety
+	var/aura_type
```

Comment:
consider making that a string as you could be emitting multiple type at once.

Reply:
there's no instance right now where you can have multiple at once and they'd have to have the same strength, but that's probably fine and wouldn't hurt to have anyway

Reply:
Isn't a phero emitter that is also a leader emitting 2?

Reply:
that's more like the queen emitting once from the leader and the leader emitting once. the two pheros don't share strength, range, or whether they're active or not